File: code/ch03/3.3.1.isPalindrome.py
class Solution:
    def isPalindrome(self, x: int) -> bool:
        if x < 0:
            return False
        if x == 0:
            return True
        if x % 10 == 0:
            return False

        res = 0
        copy = x
        while x > 0:
            res = res * 10 + (x % 10)
            x = x // 10

        return copy == res

File: code/ch03/3.3.1.isPalindrome.cpp
class Solution {
public:
  bool isPalindrome(int x) {
    if (x < 0) return false;
    if (x == 0) return true;
    if (x % 10 == 0) return false;
    int res = 0, copy = x;
    while (x > 0) {
      res = res * 10 + (x % 10);
      x /= 10;
    }
    return copy == res;
  }
};
